posthog/models/feature_flag/flag_matching.py

```diff
@@ -129,6 +133,20 @@ def get_match(self, feature_flag: FeatureFlag) -> FeatureFlagMatch:
 
         highest_priority_evaluation_reason = FeatureFlagMatchReason.NO_CONDITION_MATCH
         highest_priority_index = 0
+
+        # Match for boolean super condition first
+        if feature_flag.filters.get("super_groups", None):
+            super_condition_value = self._super_condition_value(feature_flag)
+
+            if super_condition_value is not None:
+                payload = self.get_matching_payload(super_condition_value, None, feature_flag)
+                return FeatureFlagMatch(
+                    match=super_condition_value,
+                    reason=FeatureFlagMatchReason.SUPER_CONDITION_MATCH,
```

Comment:
reason can be `SUPER_CONDITION_NOT_MATCH` as well, depending on `super_condition_value` ?

Or actually, we could infer that based on the value and keep the reason singular.

Reply:
maybe `SUPER_CONDITION_VALUE`?

Reply:
yeah better! In either case, we'd need to update the frontend later on to express what this reason means.